test/e2e/dns_autoscaling.go

```diff
@@ -207,14 +226,16 @@ func deleteDNSAutoscalerPod(c clientset.Interface) error {
 	return nil
 }
 
-func waitForDNSReplicasSatisfied(c clientset.Interface, expected int, timeout time.Duration) (err error) {
+func waitForDNSReplicasSatisfied(c clientset.Interface, getExpected getExpectReplicasFunc, timeout time.Duration) (err error) {
 	var current int
+	expected := getExpected(c)
 	framework.Logf("Waiting up to %v for kube-dns reach %v replicas", timeout, expected)
```

Comment:
why not log this in the condition func? otherwise we'll just get a few minuts of silence with no clarity into what the test is doing right.